### PR TITLE
[enhance](catalog)External partition prune return partitionName instead of partitionId

### PR DESCRIPTION
**Pull request by zddr**
### What problem does this PR solve?
The partition ID of external data sources is meaningless, and some data sources only have partition names, so the return result of partition pruning is replaced with name instead of ID
Issue Number: close #xxx

Related PR: #xxx

Problem Summary:
External partition prune return partitionName instead of partitionId
### Release note

External partition prune return partitionName instead of partitionId

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [x] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OneListPartitionEvaluator.java [hunk 1]
```diff
@@ -35,25 +35,25 @@
 import java.util.stream.IntStream;
 
 /** OneListPartitionInputs */
-public class OneListPartitionEvaluator
-        extends DefaultExpressionRewriter<Map<Slot, PartitionSlotInput>> implements OnePartitionEvaluator {
-    private final long partitionId;
+public class OneListPartitionEvaluator<K>
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OneListPartitionEvaluator.java:38]: Why add this <K>， this seems not used by this pr

Comment (zddr) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/OneListPartitionEvaluator.java:38]: The ID used for partitioning the inner table, the name used for the external table, and the underlying clipping method are compatible with generics

**Comment by zddr** [comment]
run p0

**Comment by zddr** [comment]
run cloud_p0

**Review by seawinde — APPROVED** [review]

**Review by feiniaofeiafei — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**